### DISCUSSION TITLE
Infinitesimal model and locations of QTL and SNP markers

### DISCUSSION THREAD

**Discussion by MattKinghorn** [gaynorr/AlphaSimR#212] [General]
Hi @gaynorr,

If I have the following: 

# Founder Genomes
nInd = 2000
nChr = 30
nSnp = 1600
nQtl = 860

founderGenomes = runMacs(nInd = nInd,
                           nChr = nChr,
                          segSites = nSnp + nQtl,
                          species = "CATTLE",
                          nThreads = 5)

# Global Parameters
SP$addTraitA(nQtlPerChr = nQtl,
             mean = means, 
             var = vars,
             corA = cors)

If I make nQltperChr equal to the number of SegSites would this be equivalent to an infinitesimal model where each SNP will then have a small effect? 

Secondly, if I have a total of 1600 SNP per chromosome, and then in the global parameters assume that nQtlPerChr = nQtl (860) does this imply that the 860 QTL chosen at random are causal? 

Thanks in advance

**Comment by gregorgorjanc** [comment]
@MattKinghorn infinitesimal model is a bit of an enigma. It’s saying that traits are influenced by soo many (infinitely many) loci that the each of each is soo small (infinitely small), hence a change in the trait will not make any/much difference in frequency of alleles at these loci. We know that this cannot hold exactly because the number of causal loci is limited, though could be very large. We also know that allele frequencies of alleles at many causal loci change. So, pedantically speaking you can not simulate an infinitesimal model in AlphaSimR. But you can get close by choosing many causal loci for your trait. The more the better, though better in what sense depends on what you are interested in.

**Comment by MattKinghorn** [comment]
@gregorgorjanc thank you for quick response. That answers my question.

A follow up question that I have is regarding SP$addSnpChip. If I assume the following:

SP$addTraitA(nQtlPerChr = 860,
mean = means,
var = vars,
corA = cors)

SP$addSnpChip(1600) 

Is there any way to see which of the 860 causal loci made it onto the designated SNP chip?

**Comment by gregorgorjanc** [comment]
@MattKinghorn have a look at this

```
library(AlphaSimR)
founderPop = quickHaplo(nInd=10, nChr=1, segSites=110)
SP = SimParam$new(founderPop)
SP$addTraitA(5)
SP$addSnpChip(5)
SP$traits
SP$snpChips
```

Which for the trait gives

```
> SP$traits
[[1]]
An object of class "TraitA"
Slot "addEff":
[1] -0.8517862  0.9159558 -0.3571021 -0.1317103
[5]  1.9512623

Slot "intercept":
[1] 0.03453588

Slot "nLoci":
[1] 5

Slot "lociPerChr":
[1] 5

Slot "lociLoc":
[1]  38  41  46  74 101

Slot "name":
[1] "Trait1"
```

and for the SNP chip gives

```
> SP$snpChips
[[1]]
An object of class "LociMap"
Slot "nLoci":
[1] 5

Slot "lociPerChr":
[1] 5

Slot "lociLoc":
[1]  2  4 29 71 79

Slot "name":
[1] "Chip1"
```

**Comment by gregorgorjanc** [comment]
@MattKinghorn can you change the discussion title to `Infinitesimal model and locations of QTL and SNP markers` to reflect what we discussed here?

**Reply by MattKinghorn** [reply]
@gregorgorjanc of course, thank you so much for your help.
